test/arithmetic/multiplier_test.dart

```diff
@@ -15,28 +15,69 @@ import 'package:rohd_hcl/src/arithmetic/evaluate_compressor.dart';
 import 'package:rohd_hcl/src/arithmetic/partial_product_sign_extend.dart';
 import 'package:test/test.dart';
 
+/// The following routines are useful only during testing
+extension TestMultiplierSignage on Multiplier {
+  /// Return true if multiplicand [a] is truly signed (fixed or runtime)
+  bool isSignedMultiplicand() => (selectSignedMultiplicand == null)
+      ? signedMultiplicand
+      : selectSignedMultiplicand!.value.isZero;
+
+  /// Return true if multiplier [b] is truly signed (fixed or runtime)
+  bool isSignedMultiplier() => (selectSignedMultiplier == null)
+      ? signedMultiplier
+      : selectSignedMultiplier!.value.isZero;
+
+  /// Return true if accumulate result is truly signed (fixed or runtime)
+  bool isSignedResult() => isSignedMultiplicand() | isSignedMultiplier();
+}
+
+/// The following routines are useful only during testing
+extension TestMultiplierAccumulateSignage on MultiplyAccumulate {
+  /// Return true if multiplicand [a] is truly signed (fixed or runtime)
+  bool isSignedMultiplicand() => (selectSignedMultiplicand == null)
+      ? signedMultiplicand
+      : !selectSignedMultiplicand!.value.isZero;
+
+  /// Return true if multiplier [b] is truly signed (fixed or runtime)
+  bool isSignedMultiplier() => (selectSignedMultiplier == null)
+      ? signedMultiplier
+      : !selectSignedMultiplier!.value.isZero;
+
+  /// Return true if addend [c] is truly signed (fixed or runtime)
+  bool isSignedAddend() => (selectSignedAddend == null)
+      ? signedAddend
+      : !selectSignedAddend!.value.isZero;
+
+  /// Return true if accumulate result is truly signed (fixed or runtime)
+  bool isSignedResult() =>
+      isSignedAddend() | isSignedMultiplicand() | isSignedMultiplier();
+}
+
 /// Simple multiplier to demonstrate instantiation of CompressionTreeMultiplier
 class SimpleMultiplier extends Module {
   /// The output of the simple multiplier
   late final Logic product;
 
   /// Construct a simple multiplier with runtime sign operation
-  SimpleMultiplier(Logic a, Logic b, Logic multASigned)
+  SimpleMultiplier(
```

Comment:
is this `SimpleMultiplier` used anywhere? should it be?

Reply:
I was using it to catch trace errors in instantiation.  I wrapped it in a test that helps to put other multipliers in the test harness that expects a MultiplyAccumulate wrapper.